pyirf/scripts/make_DL3.py

```diff
@@ -0,0 +1,338 @@
+            args.pipeline,
+            # cfg["analysis"]["thsq_opt"]["type"], # not needed now
+            # cfg["analysis"]["obs_time"]["value"],
+            # cfg["analysis"]["obs_time"]["unit"],
+            args.obs_time.split(".")[0],
+            args.obs_time.split(".")[-1],
+        ),
+    )  # and create it if necessary
+    if not os.path.exists(outdir):
+        os.makedirs(outdir)
+
+    # =========================================================================
+    #                   READ DL2 DATA AND STORE IT ACCORDING TO GADF
+    # =========================================================================
+
+    # Load FITS data
+    particles = ["gamma", "electron", "proton"]
+    evt_dict = dict()  # Contain DL2 file for each type of particle
+    for particle in particles:
+        if args.debug:
+            print(f"Loading {particle} DL2 data...")
+        infile = os.path.join(indir, template_input_file.format(particle))
+        evt_dict[particle] = read_FITS(config=cfg, infile=infile)
+
+    # =========================================================================
+    #               PRELIMINARY OPERATIONS FOR SPECIFIC PIPELINES
+    # =========================================================================
+
+    # Some pipelines could provide some of the DL2 data in different ways
+    # After this part, DL2 data is supposed to be equivalent, regardless
+    # of the original pipeline.
+
+    # Later we should move this out of here, perhaps under a "utils" module.
+
+    if args.pipeline == "EventDisplay":
+
+        # THETA IS SUPPOSED TO BE IN DL2 DATA ACCORDING TO GADF
+        # EventDisplay provides true and reconstructed directions, so we
+        # calculate it here and we add it to the tables.
+
+        # WARNING:
+        # this is true only for point-source simulations!
+        # in the case of diffuse simulations we must distinguish between
+        # - distance between reconstructed direction and center of the FoV
+        # - distance between reconstructed and true direction
+
+        for particle in particles:
+
+            THETA = angular_separation(
+                evt_dict[particle]["TRUE_AZ"],
+                evt_dict[particle]["TRUE_ALT"],
+                evt_dict[particle]["AZ"],
+                evt_dict[particle]["ALT"],
+            )  # in degrees
+
+            # Add THETA column
+            evt_dict[particle]["THETA"] = THETA
+
+            # print(evt_dict[particle].head(n=5))
+
+    # =========================================================================
+    #                   REST OF THE OPERATIONS (TO BE REFACTORED)
+    # =========================================================================
+
+    # Apply offset cut to proton and electron
+    for particle in ["electron", "proton"]:
+
+        # There seems to be a problem in using pandas from FITS data
+        # ValueError: Big-endian buffer not supported on little-endian compiler
```

Comment:
Yes, you need to convert to native byteorder like so: https://github.com/fact-project/pyfact/blob/65201e6aa029264b0b4305c6bc8731b3a11b79cc/fact/io.py#L127

Reply:
but using astropy.table.Table.to_pandas should do that also for you.